registry/storage/ociartifactmanifesthandler.go

```diff
@@ -112,3 +122,32 @@ func (ms *ociArtifactManifestHandler) verifyArtifactManifest(ctx context.Context
 
 	return nil
 }
+
+// indexReferrers indexes the subject of the given revision in its referrers index store.
+func (ms *ociArtifactManifestHandler) indexReferrers(ctx context.Context, dm *ociartifact.DeserializedManifest, revision digest.Digest) error {
+	if dm.Subject == nil {
+		return nil
+	}
+
+	// [TODO] We can use artifact type in the link path to support filtering by artifact type
+	//  but need to consider the max path length in different os
+	subjectRevision := dm.Subject.Digest
+
+	rootPath := path.Join(referrersLinkPath(ms.repository.Named().Name()), subjectRevision.Algorithm().String(), subjectRevision.Hex())
+	referenceLinkPath := path.Join(rootPath, revision.Algorithm().String(), revision.Hex(), "link")
+	if err := ms.storageDriver.PutContent(ctx, referenceLinkPath, []byte(revision.String())); err != nil {
+		return err
+	}
+	return nil
+}
+
+// // TODO: Should be removed and paths package used
+func referrersRepositoriesRootPath(name string) string {
+	return path.Join(referrersStorageRootPath, "repositories", name)
+}
+
+// TODO: Should be removed and defined instead in paths package
+// Requires paths package to be exported
+func referrersLinkPath(name string) string {
```

Comment:
Need advice on how to handle these.

Reply:
@wangxiaoxuan273 Yes, please move them to the `path` package and update the related documentation.

Reply:
Done.